src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPCompletionFeature.java

```diff
@@ -220,27 +220,38 @@ public boolean isItemTextBold(@NotNull CompletionItem item) {
      * @param lookupItem
      * @param priority
      * @param item
+     * @param caseSensitive
      */
     @ApiStatus.Internal
     public void addLookupItem(@NotNull CompletionPrefix completionPrefix,
                               @NotNull CompletionResultSet result,
                               @NotNull LookupElement lookupItem,
                               int priority,
-                              @NotNull CompletionItem item) {
+                              @NotNull CompletionItem item,
+                              boolean caseSensitive) {
         var prioritizedLookupItem = PrioritizedLookupElement.withPriority(lookupItem, priority);
         // Compute the prefix
         var textEditRange = ((LSPCompletionProposal) lookupItem).getTextEditRange();
         String prefix = textEditRange != null ? completionPrefix.getPrefixFor(textEditRange, item) : null;
         if (prefix != null) {
-            // Add the IJ completion item (lookup item) by using the computed prefix
-            result.withPrefixMatcher(prefix)
-                    .caseInsensitive()
-                    .addElement(prioritizedLookupItem);
+            // Add the IJ completion item (lookup item) by using the computed prefix respecting the language's case-sensitivity
+            if (caseSensitive) {

```

Comment:
I could condense these two conditionals slightly be extracting `result.withPrefixMatcher()` into a variable and setting other state conditionally, but I didn't find that quite as readable.